lib/libzfs/libzfs_pool.c

```diff
@@ -241,6 +241,38 @@ zpool_pool_state_to_name(pool_state_t state)
 	return (gettext("UNKNOWN"));
 }
 
+/*
+ * Given a pool handle, return the pool health string ("ONLINE", "DEGRADED",
+ * "SUSPENDED", etc).
+ */
+const char *
+zpool_get_health_str(zpool_handle_t *zhp)
+{
+	zpool_errata_t errata;
+	zpool_status_t status;
+	nvlist_t *nvroot;
+	vdev_stat_t *vs;
+	uint_t vsc;
+	const char *str;
+
+	status = zpool_get_status(zhp, NULL, &errata);
+
+	if (zpool_get_state(zhp) == POOL_STATE_UNAVAIL) {
+		str = gettext("FAULTED");
+	} else if (status == ZPOOL_STATUS_IO_FAILURE_WAIT ||
+	    status == ZPOOL_STATUS_IO_FAILURE_MMP) {
+		str = gettext("SUSPENDED");
```

Comment:
since suspended state is in addition to other states, do we need a separate kstat for suspension?

Reply:
I don't think so.  The kstat is there to be a quick "is the pool healthy?" check for pacemaker to use.  If you wanted to know why the pool was suspended, you could always look at the `zpool status` output.

Reply:
> the zpool command suspension message is sent to stderr, so most screen-scrapers miss it.

@richardelling which `zpool status` suspension message are you referring too?  My recollection is that this has always been a point of confusion for users because a pool may be suspended yet appear totally healthy in the `zpool status` output.

Reply:
yes, it is confusing :-(  `zfs_standard_error[_fmt]()` has many callers and buried in there is the case where users are notified over stderr that the pool is suspended. In any case, I think exposing spa_suspended can be useful, but belongs in another PR